driver-core/src/main/com/mongodb/client/model/WindowedComputations.java

```diff
@@ -0,0 +1,772 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model;
+
+import com.mongodb.client.model.Windows.Bound;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentWriter;
+import org.bson.BsonType;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.client.model.MongoTimeUnit.MONTH;
+import static com.mongodb.client.model.MongoTimeUnit.YEAR;
+import static com.mongodb.client.model.WindowedComputations.ParamName.ALPHA;
+import static com.mongodb.client.model.WindowedComputations.ParamName.BY;
+import static com.mongodb.client.model.WindowedComputations.ParamName.DEFAULT;
+import static com.mongodb.client.model.WindowedComputations.ParamName.INPUT;
+import static com.mongodb.client.model.WindowedComputations.ParamName.N;
+import static com.mongodb.client.model.WindowedComputations.ParamName.OUTPUT;
+import static com.mongodb.client.model.WindowedComputations.ParamName.UNIT;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Builders for {@linkplain WindowedComputation windowed computations} used in the
+ * {@link Aggregates#setWindowFields(Object, Bson, List) $setWindowFields} pipeline stage
+ * of an aggregation pipeline. Each windowed computation is a triple:
+ * <ul>
+ *     <li>A window function. Some functions require documents in a window to be sorted
+ *     (see {@code sortBy} in {@link Aggregates#setWindowFields(Object, Bson, List)}).</li>
+ *     <li>An optional {@linkplain Window window}, a.k.a. frame.
+ *     Specifying {@code null} window is equivalent to specifying [{@link Bound#UNBOUNDED}, {@link Bound#UNBOUNDED}].</li>
+ *     <li>A path to an output field to be computed by the window function over the window.</li>
+ * </ul>
+ * A windowed computation is similar to an {@linkplain Accumulators accumulator} but does not result in folding documents constituting
```

Comment:
While `BsonField`s representing windowed computations are similar to those produced by the `Accumulators` methods, they are different at least because the field value documents include windows. Thus, we need a new set of builders, which are provided by the `WindowedComputations` class.